### DISCUSSION TITLE
Difference from a single atom in reagent?

### DISCUSSION THREAD

**Discussion by NOBLES5E** [day8/re-frame#687] [Q&A] [answered]
Sorry for my noob question. It seems to me that I can do what re-frame does in reagent by defining a single global atom like this:

```clojure
;; data store
(def global-store (r/atom {... initial map ...}))

;; event/dispatcher
(defn change-page [page] (swap! global-store update :page page))

;; view
(defn webpage [] [:button {:on-click #(change-page "clicked")} (:page @global-store)])
```

I know there must be something I am missing here. Could someone explain a bit on what advantages re-frame has over just using reagent? I googled and did not find an article on this.. Thanks

**Comment by gadfly361** [comment, marked as answer]
@ikzk Hey, glad you have found re-frame! :)

Here is an [answer to your question](https://github.com/Day8/re-frame/blob/master/docs/FAQs/DoINeedReFrame.md).

As an aside, for future questions like this, there is a very friendly community on [Clojure Slack](http://clojurians.net) in the #re-frame channel.

**Comment by NOBLES5E** [comment]
@gadfly361 Thanks for your very quick response!
